corehq/apps/couch_sql_migration/couchsqlmigration.py

```diff
@@ -236,15 +236,28 @@ def _migrate_form_and_associated_models_async(self, wrapped_form):
             self.processed_docs += 1
             self._log_main_forms_processed_count(throttled=True)
 
-    def _migrate_form_and_associated_models(self, couch_form):
-        sql_form = _migrate_form(self.domain, couch_form)
+    def _migrate_form_and_associated_models(self, couch_form, form_is_processed=True):
+        """
+        Copies `couch_form` into a new sql form
+        """
+        if form_is_processed:
```

Comment:
Have you tried removing it and see if the tests pass? I'm not sure if that's sufficient verification to conclude that it _is not_ necessary, but it may answer the question of whether it _is_ necessary (and hopefully "why" too).

Reply:
If I drop the `if` clause and just keep the code for the `True` condition, the tests pass.

Reply:
Hmm, so that tells us that either it's fine or there are no tests that cover the situation that this was written for (which seems plausible). It's hard to say if it's safe to remove this.

Reply:
... pass, or don't run locally but do run on Travis. I'll revert dropping `form_is_processed`.